### DISCUSSION TITLE
Availability of finetuning for new datasets?

### DISCUSSION THREAD

**Discussion by mengziwen2000** [neuralhydrology/neuralhydrology#151] [Q&A] [answered]
Can fine-tuning of watersheds outside the CAMELS_US dataset be done on neuralhydrology using CAMELS_US as the source dataset?

**Comment by mengziwen2000** [comment]
You say, "Our target basin does not necessarily have to be a part of this dataset, but usually it should be better to include it." , now I set the `dataset` to `generic`, then my time series (.nc file) is in the time_series folder, but it always says ' `KeyError: "None of [Index(['prcp', 'tmax', 'tmin', 'surf_pres', 'QObs']. dtype='object')] are in the [columns]"` 'Error. I've implemented a way to set the dataset to generic, indicating that my nc file is not wrong. So I don't really know what's wrong.

**Reply by kratzert** [reply, marked as answer]
Sure this is possible. If you don't want to touch code, you need to make sure though that the features in your new dataset, which you want to use for finetuning, have the same names as the CAMELS-US features you use for training. Everything else requires minor changes to the code.

**Reply by mengziwen2000** [reply]
Okay, thanks, I'll give it a try

**Comment by mengziwen2000** [comment]
Hi, now, both methods implement rainfall runoff prediction for the new dataset, one will be set `generic` and the other will process the data into the format of the CAMELS_US dataset, which suggests that the data is fine. However, when fine-tuning the new dataset generated by this latter method, an error like` ' RuntimeError: Loss was NaN for 1 times in a row. stopped training.' `still occurs. What do you think is causing this ?

**Reply by kratzert** [reply]
Not really sure I can follow. You finetune to your own dataset, not CAMELS_US right? Also, really hard to solve this without seeing the config you use for training and the config you use for finetuning (as well as maybe a data example for your fine-tuning dataset)

**Reply by mengziwen2000** [reply]
Yes, I am fine-tuning to my own dataset, pre-training with CAMELS_US, and the data examples (or feature names) used to fine-tune the dataset are the same as CAMELS_US. The configuration used for training and the configuration used for fine-tuning is consistent with finetuning.ipynb. Could this be why I placed the dataset used for fine-tuning and the CAMELS_US dataset together? But I don't think it is, so maybe I can try that first. If this route works, wow, migration learning will be easy under neuralhydrology. It's still quite interesting!

**Reply by mengziwen2000** [reply]
Pre-training with the first 200 watersheds of CAMLES_US and fine-tuning the 201st watershed with all the config files I have set up unchanged, and this worked. So, there should be something wrong with my own dataset settings. I should check it properly again. Thanks.
